lib/spack/spack/cmd/load.py

```diff
@@ -4,22 +4,65 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 import argparse
-from spack.cmd.common import print_module_placeholder_help, arguments
+import sys
 
-description = "add package to environment using `module load`"
-section = "modules"
+import llnl.util.tty as tty
+
+import spack.cmd
+from spack.cmd.common import arguments
+import spack.environment as ev
+import spack.util.environment
+import spack.user_environment as uenv
+
+description = "add package to the user environment variables"
+section = "user environment"
 level = "short"
 
 
 def setup_parser(subparser):
     """Parser is only constructed so that this prints a nice help
        message with -h. """
+    shells = subparser.add_mutually_exclusive_group()
+    shells.add_argument(
+        '--sh', action='store_const', dest='shell', const='sh',
+        help="print sh commands to load the package")
+    shells.add_argument(
+        '--csh', action='store_const', dest='shell', const='csh',
+        help="print csh commands to load the package")
+
     subparser.add_argument(
-        'spec', nargs=argparse.REMAINDER,
-        help="spec of package to load with modules "
+        'specs', nargs=argparse.REMAINDER,
+        help="spec of package to load"
     )
     arguments.add_common_arguments(subparser, ['recurse_dependencies'])
 
 
 def load(parser, args):
-    print_module_placeholder_help()
+    env = ev.get_env(args, 'load')
+    specs = list(map(lambda spec: spack.cmd.disambiguate_spec(spec, env),
+                     spack.cmd.parse_specs(args.specs)))
+    if not args.shell:
+        msg = [
+            "This command works best with Spack's shell support",
+            ""
+        ] + spack.cmd.common.shell_init_instructions + [
+            'Or, if you want to use `spack load` without initializing',
+            'shell support, you can run one of these:',
+            '',
+            '    eval `spack load --sh %s`   # for bash/sh' % args.specs,
+            '    eval `spack load --csh %s`  # for csh/tcsh' % args.specs,
+        ]
+        tty.msg(*msg)
+        return 1
+
+    if args.recurse_dependencies:
```

Comment:
The `-r` option is always default False on other packages.

I think it will be better to replace it with a `--only` argument like the install command has. Default will be package and dependencies, but `--only` can specify only dependencies or only the package itself.

Reply:
I like this idea.  `--only` does more and has the right default.